packages/sol6/src/yang/etsi-nfv-ns.yang

```diff
@@ -758,101 +842,106 @@ submodule etsi-nfv-ns {
 
             description
               "Defines the connection information of the VNF, it
-               contains connection relationship between a VNF
-               connection point and a NS Virtual Link.";
+              contains connection relationship between a VNF
+              connection point and a NS Virtual Link.";
             reference
               "GS NFV IFA014: Section 6.3.3.2 VnfProfile information
-               element";
-
-	    leaf virtual-link-profile-id {
-	      type leafref {
-		path "../../../virtual-link-profile/id";
-	      }
-	      description
-		"Reference an NS VL profile.";
-	      reference
-		"GS NFV IFA014: Section 6.3.7.2, NsVirutalLinkConnectivity
-                 information element.";
-	    }
-
-	    leaf-list cpd-id {
-	      type leafref {
-		path "deref(../../vnfd-id)/../ext-cpd/id";
-	      }
-	      description
-		"References the descriptor of a connection point on a
-                 VNF/PNF or a SAP which connects to virtual links
-                 instantiated from the profile identified in the
-                 virtualLinkProfileId attribute.";
-	      reference
-		"GS NFV IFA014: Section 6.3.7.2, NsVirutalLinkConnectivity
-                 information element.";
-	    }
+              element";
+
+            leaf virtual-link-profile-id {
+              type leafref {
+                path "../../../virtual-link-profile/id";
+              }
+              description
+                "Reference an NS VL profile.";
+              reference
+                "GS NFV IFA014: Section 6.3.7.2, NsVirutalLinkConnectivity
+                information element.";
+            }
+
+            leaf-list cpd-id {
+              type leafref {
+                path "deref(../../vnfd-id)/../ext-cpd/id";
+              }
+              description
+                "References the descriptor of a connection point on a
+                VNF/PNF or a SAP which connects to virtual links
+                instantiated from the profile identified in the
+                virtualLinkProfileId attribute.";
+              reference
+                "GS NFV IFA014: Section 6.3.7.2, NsVirutalLinkConnectivity
+                information element.";
+            }
           }
         }
 
         list pnf-profile {
           key "id";
+          must "boolean(../../pnfd-id[.=current()/pnfd-id])" {
+            error-message "PNFDs in the profile has to be listed as a " +
```

Comment:
I love error-message, and would like to see more of these. Is there a description and reference that we can add? Also, can we collapse the next two lines into one.